java-security/src/main/java/com/sap/cloud/security/adapter/xs/XSUserInfoAdapter.java

```diff
@@ -318,16 +357,77 @@ private String getClaimValue(String claimname) throws XSUserInfoException {
 	}
 
 	@Nullable
-	private JsonObject getClaimAsJsonObject(String claimName) throws XSUserInfoException {
+	private JsonObject getClaimAsJsonObject(String claimName) {
 		try {
 			return accessToken.getClaimAsJsonObject(claimName);
 		} catch (JsonParsingException e) {
 			throw createXSUserInfoException(claimName).get();
 		}
 	}
 
-	String getExternalAttribute(String attributeName) throws XSUserInfoException {
+	String getExternalAttribute(String attributeName) {
 		return getAttributeFromClaimAsString(EXTERNAL_ATTRIBUTE, attributeName);
 	}
 
+	private String performTokenFlow(String baseUrl, int tokenRequestType, String clientId, String clientSecret,
+			Map<String, String> additionalAuthAttributes) {
+		try {
+			ClientCredentials clientCredentials = new ClientCredentials(clientId, clientSecret);
+			XsuaaTokenFlows xsuaaTokenFlows = new XsuaaTokenFlows(oAuth2TokenService,
```

Comment:
raise UnsupportedOperationException in case oAuth2TokenService is null... furthermore allow a setter

Reply:
Alright could make sense to override that for some applications and then we cannot control and should check for `null`, I will do that.